### PR TITLE
Creates nmap cheat sheet

### PR DESCRIPTION
**Pull request by aleksandar-todorovic**
**What does your Instant Answer do?**
Lists `nmap`'s basic and advanced options for executing `nmap` from the command line.

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://nmapcookbook.blogspot.com.br/2010/02/nmap-cheat-sheet.html

**Are there any other alternative (better) data sources?**
There's a `.pdf` file with the same content.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Security researchers

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**
https://duck.co/ideas/idea/6265/nmap-cheat-sheet-commands

**Are you having any problems? Do you need our help with anything?**
Nope, everything's working great. I think it's ready to be merged. Although I could have made a typo somewhere.

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screenshot from 2015-08-29 17 01 14

IA Page: https://duck.co/ia/view/nmap_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/nmap.json [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+            "key": "$ nmap -sF \\[target\\]"
+        }, {
+            "val": "Xmas scan",
+            "key": "$ nmap -sA \\[target\\]"
+        }, {
+            "val": "TCP ACK scan",
+            "key": "$ nmap -sA \\[target\\]"
+        }, {
+            "val": "Custom TCP scan",
+            "key": "$ nmap --scnaflags \\[flags\\] \\[target\\]"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/nmap.json]: should that be `--scanflags` ?

**Comment by mintsoft** [comment]
@aleksandar-todorovic This looks pretty good to me, thanks for the contribution. I've put some minor things against the diff that I noticed :)

File: share/goodie/cheat_sheets/json/nmap.json (hunk 1 above)
Comment (aleksandar-todorovic) [inline share/goodie/cheat_sheets/json/nmap.json]: Yup. I also noticed one more typo that will be fixed in like 10 seconds.

**Comment by jagtalon** [comment]
@mintsoft would something more "official" like http://linuxcommand.org/man_pages/nmap1.html be better?

**Comment by mintsoft** [comment]
@jagtalon Absolutely

**Comment by aleksandar-todorovic** [comment]
Judging by the [.pdf file](http://hackingdefined.org/publications/nmap-cheat-sheet.pdf) I mentioned that contains the same content:

> This nmap cheat sheet is uniting a few other cheat sheets

...whatever that means.

**Comment by mintsoft** [comment]
@aleksandar-todorovic if all the content is covered by the manpage (I imagine that it is) then it's the best possible source for the IA

File: share/goodie/cheat_sheets/json/nmap.json [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        "Firewall Evasion Techniques",
+        "Output Options",
+        "Troubleshooting and Debugging",
+        "Nmap Scripting Engine",
+        "Ndiff"
+    ],
+    "sections": {
+        "Basic Scanning Techniques": [{
+            "val": "Scan a single target",
+            "key": "$ nmap \\[target\\]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/nmap.json]: let's remove all the leading `$ ` -- It should be clear from the template that we're showing a terminal prompt

File: share/goodie/cheat_sheets/json/nmap.json [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+            "val": "Service version detection",
+            "key": "$ nmap -sV \\[target\\]"
+        }, {
+            "val": "Troubleshooting version scans",
+            "key": "$ nmap -sV --version-trace \\[target\\]"
+        }, {
+            "val": "Perform a RPC scan",
+            "key": "$ nmap -sR \\[target\\]"
+        }],
+        "Timing options": [{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/nmap.json]: typo: need to capitalize `Options`

File: share/goodie/cheat_sheets/json/nmap.json [hunk 4]
```diff
@@ -0,0 +1,326 @@
+{
+    "id": "nmap_cheat_sheet",
+    "name": "nmap",
+    "description": "Nmap Security Scanner",
+    "metadata": {
+        "sourceName": "nmap cookbook",
+        "sourceUrl": "http://nmapcookbook.blogspot.com.br/2010/02/nmap-cheat-sheet.html"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/nmap.json]: Are we in agreement about the source then? Do we know that all of these commands work and exist in the man page? 

/cc @mintsoft

**Comment by moollaza** [comment]
@aleksandar-todorovic this is **nearly** done! There's a couple small changes necessary and then this is good for merge :+1:

File: share/goodie/cheat_sheets/json/nmap.json (hunk 4 above)
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/nmap.json]: Looks like it yeah

**Comment by aleksandar-todorovic** [comment]
Oops, looks like I missed the last notification. @moollaza thanks for pinging me! The latest commit removes the `$` sign and capitalizes the letter `o` in `Timing Options`.

File: share/goodie/cheat_sheets/json/nmap.json [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        }, {
+            "val": "Troubleshoot scripts",
+            "key": "nmap --script \\[script\\] --script-trace \\[target\\]"
+        }, {
+            "val": "Update the script database",
+            "key": "nmap --script-updatedb"
+        }],
+        "Ndiff": [{
+            "val": "Comparison using Ndiff",
+            "key": "$ ndiff \\[scan1.xml\\] \\[scan2.xml\\]"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/nmap.json]: Still a few `$` to be removed. This is good to merge though :+1:

**Comment by moollaza** [comment]
@aleksandar-todorovic this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@aleksandar-todorovic congrats! It's live: https://duckduckgo.com/?q=nmap+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**